pkg/kubelet/cm/container_manager_linux_test.go

```diff
@@ -342,3 +360,230 @@ func TestNewPodContainerManager(t *testing.T) {
+			cpuSets:                         map[string]cpuset.CPUSet{"pod1/c1": cpuset.New(1)},
+			expectExclusiveCPUs:             true,
+			podLevelResourceManagersEnabled: true,
+		},
+		{
+			name: "Pod-level resources, Guaranteed, Integer CPUs, no exclusive CPUs assigned",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{UID: "pod1"},
+				Spec: v1.PodSpec{
+					Resources:  guaranteedPodResources,
+					Containers: []v1.Container{{Name: "c1", Resources: guaranteedContainerResources}},
+				},
+			},
+			containerName:                   "c1",
+			cpuSets:                         map[string]cpuset.CPUSet{},
+			expectExclusiveCPUs:             false,
+			podLevelResourceManagersEnabled: true,
+		},
+		{
+			name: "Pod-level resources, Burstable container",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{UID: "pod1"},
+				Spec: v1.PodSpec{
+					Resources:  guaranteedPodResources,
+					Containers: []v1.Container{{Name: "c1", Resources: burstableContainerResources}},
+				},
+			},
+			containerName:                   "c1",
+			cpuSets:                         map[string]cpuset.CPUSet{"pod1/c1": cpuset.New(1)},
+			expectExclusiveCPUs:             false,
+			podLevelResourceManagersEnabled: true,
+		},
+		{
+			name: "Pod-level resources, Guaranteed, non-integer CPU",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{UID: "pod1"},
+				Spec: v1.PodSpec{
+					Resources:  guaranteedPodResources,
+					Containers: []v1.Container{{Name: "c1", Resources: nonIntegerContainerResources}},
+				},
+			},
+			containerName:                   "c1",
+			cpuSets:                         map[string]cpuset.CPUSet{"pod1/c1": cpuset.New(1)},
+			expectExclusiveCPUs:             false,
+			podLevelResourceManagersEnabled: true,
+		},
+		{
+			name: "Feature gate off, does not have exclusive CPUs",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{UID: "pod1"},
+				Spec: v1.PodSpec{
+					Resources:  guaranteedPodResources,
+					Containers: []v1.Container{{Name: "c1"}},
+				},
+			},
+			containerName:                   "c1",
+			cpuSets:                         map[string]cpuset.CPUSet{},
+			expectExclusiveCPUs:             false,
+			podLevelResourceManagersEnabled: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.pod.UID == "" {
+				tc.pod.UID = types.UID("pod1")
+			}
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, kubefeatures.PodLevelResourceManagers, tc.podLevelResourceManagersEnabled)
+
+			mockReader := &mockCPUAllocationReader{
+				sets: tc.cpuSets,
+			}
+			cm := &containerManagerImpl{
+				cpuManager: mockReader,
+			}
+
+			var targetContainer *v1.Container
+			for i := range tc.pod.Spec.Containers {
+				if tc.pod.Spec.Containers[i].Name == tc.containerName {
+					targetContainer = &tc.pod.Spec.Containers[i]
+					break
+				}
+			}
+			if targetContainer == nil {
+				t.Fatalf("container %s not found in pod spec", tc.containerName)
+			}
```

Comment:
Honestly I'd let the test crash if this is the case. The tests are highly controlled environment and mistakes which are checked here should be sporadic, very reproducible and caught immediately caught when adding tests. Keeping this check is not wrong, though, just feels... awkward?

Reply:
OTOH if you are concerned/have found that feeding `ContainerHasExclusiveCPUSs` with a inexistent container name causes the function to crash/misbehave, then we should rather fix the `ContainerHasExclusiveCPUs` function, e.g. adding an error return value. Should be done separately though, likely in its own PR